### PR TITLE
[project:watch] fix temp file error

### PR DESCRIPTION
**Pull request by bhageena**
The intention is to have a system-defined ignore pattern list and also allow users to specify `ignore` patterns via `.exclude` file in the project root.

WIP:
the  watcher is working (i.e. ignoring the changes in files matching exclude patterns), but the deployer `doDeploy` is still having the error, will have to work on it 
relates to #46

### CODE REVIEW

**Comment by bhageena** [comment]
Cases that were failing earlier are now working, not very extensively tested though, opening up for further discussion and review

File: deployer/src/util.ts [hunk 1]
```diff
@@ -1373,3 +1384,21 @@ export function renamePackage(spec: DeployStructure, oldName: string, newName: s
...
+// Checks if a given pattern matches exclusion list, defined by system or user via global .exclude file.
+export function isExcluded(match: string): boolean {
+  return anymatch(getExclusionList(), match)
+}
+
+// Returns full list of exclusion patterns, predefined or listed in the global .exclude file.
+export function getExclusionList(): string[] {
+  let userDefinedPatterns = []
+  try {
+    const globalExcludeFile = path.join(process.cwd(), '.exclude')
```
Comment (joshuaauerbachwatson) [inline deployer/src/util.ts:1397]: I'm concerned about the use of process.cwd() here.   I would expect to read this information either (1) from the root of the project, or (2) from the user's home directory, depending on what scope you are looking for.   I'm not sure which of those two scopes is better (let's discuss) but I'm pretty sure the current directory is not going to work in general.

File: src/commands/project/watch.ts [hunk 2]
```diff
@@ -90,7 +90,7 @@ function watch(project: string, cmdFlags: Flags, creds: Credentials|undefined, o
   }
   const watch = () => {
     // logger.log("Opening new watcher")
-    watcher = chokidar.watch(project, { ignoreInitial: true, followSymlinks: false, usePolling: false, useFsEvents: false })
+    watcher = chokidar.watch(project, { ignoreInitial: true, followSymlinks: false, usePolling: false, useFsEvents: false, ignored: getExclusionList() })
```
Comment (joshuaauerbachwatson) [inline src/commands/project/watch.ts:93]: Nice touch to use chokidar's own `ignored` flag here, but, are you sure of the semantics (I haven't used it myself).

**Review by joshuaauerbachwatson — COMMENTED** [review]
@bhageena I apologize for the delay in reviewing this.  Based on your comments I am assuming you are looking for feedback, not necessarily final signoff.   This looks very good and just what the issue is asking for.   However, I posted two comments, one of which is a serious concern, so let's discuss when you have a chance.

File: deployer/src/util.ts (hunk 1 above)
Comment (bhageena) [reply deployer/src/util.ts]: Thanks!
The project path is fed to the command, using args, so  `process.cwd()` is not the right choice here.
Agree. I think, option 1  to be the right choice.

File: src/commands/project/watch.ts (hunk 2 above)
Comment (bhageena) [reply src/commands/project/watch.ts]: I tested it, we'll have unit tests soon!

File: src/commands/project/watch.ts [hunk 3]
```diff
@@ -11,7 +11,7 @@
  * governing permissions and limitations under the License.
  */
 
-import { NimBaseCommand, NimLogger, Flags, Credentials, OWOptions, inBrowser, isGithubRef, delay } from 'nimbella-deployer'
+import { NimBaseCommand, NimLogger, Flags, Credentials, OWOptions, inBrowser, isGithubRef, delay, getExclusionList, isExcluded } from 'nimbella-deployer'
```
Comment (rabbah) [inline src/commands/project/watch.ts:14]: @bhageena check these warnings below 🙏

**Review by rabbah — COMMENTED** [review]
lgtm

**Review by joshuaauerbachwatson — APPROVED** [review]
@bhageena this looks great.   As Rodric pointed out in his review, there is a lint warning that should be eliminated (it looks like four warnings but it's really the same warning four times because of the multiple node versions).  Can you do that before I merge?   It'll save me a chore.  Thanks.

File: src/commands/project/watch.ts (hunk 3 above)
Comment (bhageena) [reply src/commands/project/watch.ts]: thanks for pointing out

**Comment by bhageena** [comment]
> @bhageena this looks great. As Rodric pointed out in his review, there is a lint warning that should be eliminated (it looks like four warnings but it's really the same warning four times because of the multiple node versions). Can you do that before I merge? It'll save me a chore. Thanks.

Sorry for the delay. I failed to realize, where the warning was till I looked closely. should I add a pre-commit hook to check linting using `husky`. It will be part of a different PR of course.

**Comment by joshuaauerbachwatson** [comment]
Yes, as we were introducing lint rules gradually, I was reluctant to use pre-commit hooks, but I think the time has come, and, if we want the hooks to run for everyone, husky is the way.  As you say, though, make that a separate PR.  I'll be merging this one.  Thanks.

**Pull request merged**